tweet-service/src/main/java/com/example/tweets/controller/RetweetController.java

```diff
@@ -0,0 +1,43 @@
+package com.example.tweets.controller;
+
+import com.example.tweets.dto.response.RetweetResponse;
+import com.example.tweets.service.RetweetService;
+import jakarta.ws.rs.QueryParam;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/retweets")
+@RequiredArgsConstructor
+public class RetweetController {
+
+    private final RetweetService retweetService;
+
+    @PostMapping("/{tweetId}")
+    public ResponseEntity<Boolean> retweet(@PathVariable Long tweetId, @RequestHeader String loggedInUser) {
+        return ResponseEntity.ok(retweetService.retweet(tweetId, loggedInUser));
+    }
+
+    @DeleteMapping("/{tweetId}")
+    public ResponseEntity<Boolean> undoRetweet(@PathVariable Long tweetId, @RequestHeader String loggedInUser) {
+        return ResponseEntity.ok(retweetService.undoRetweet(tweetId, loggedInUser));
+    }
+
+    @GetMapping("/{tweetId}")
+    public ResponseEntity<Boolean> isRetweeted(@PathVariable Long tweetId, @RequestHeader String loggedInUser) {
+        return ResponseEntity.ok(retweetService.isRetweeted(tweetId, loggedInUser));
+    }
+
+    @GetMapping
+    public ResponseEntity<RetweetResponse> getRetweet(@QueryParam(value = "retweetId") Long retweetId) {
```

Comment:
Is there a specific reason for using a query param for an id? Personally, i don't think so ...
It'd be better to include id into URL path, because we followed this approach before our whole development.
It'll make the endpoint more descriptive and aligns with RESTful principles :)